app/jobs/pubsub_job.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+class PubsubJob < ApplicationJob
+  queue_as(:default)
+  retry_on(StandardError, wait: 5.minutes, attempts: 3)
+
+  self.queue_adapter = :pubsub
+
+  def perform(data)
+    Rails.logger.info("[PubsubJob] Perform Now: #{data}")
+
+    raise(StandardError) if rand < 0.2 # fail 20% of the time
+
+    Pubsub.publish!(data)
```

Comment:
The job should dequeue from Pub/Sub, not publish the actual job to it.

Reply:
I did asked some clarification about where to put the job (enqueueing or dequeueing). It was my mistake to submit the solution without waiting for things to be clarified.

Reply:
I see. Unfortunately, I don't think I could have helped with this - it's one of the core parts of the challenge. I would have given away too much.

Reply:
Understood. No worries!